rel/files/mongooseim.toml

```diff
@@ -0,0 +1,362 @@
+[general]
+  loglevel = 3
+  hosts = [
+    "localhost",
+    "anonymous.localhost",
+    "localhost.bis"
+  ]
+  registration_timeout = "infinity"
+  language = "en"
+  all_metrics_are_global = false
```

Comment:
> In this review, I focused mostly on the TOML file. Below there are some more general comments:
> 
>     * If possible, I'd like to see `modules` and `services` section moved up (modules before services would be perfect). They change more often than `shapers` and `acls`.
I will experiment with the order for a moment and see how it looks like.

>     * Below there are config options which, in the following changes, could be removed from the TOML config file and exchanged with some kind of defaults (if not present in the config):
>       
>       * `registration_timeout`
>       * `language`
>       * `all_metrics_are_global`
>       * `max_fsm_queue`
I think that introducing the new defaults is a must and should be done in a new task.

> When it comes to the code parsing and translating config option to the intermediate format it could be refactored in a way that some operations are delegated to modules responsible for certain functionality.
Yes, I fully agree with that.

> Also, I think in many cases the translation from binary to list is redundant, because later the list is often converted to binaries again. This would require changing the intermediate format, which I'm not sure we are going to do in the nearest future.
Again, we think the same :-) it just became more obvious now how we convert things there and back. I left it as it is on purpose to limit the changes in one go. We definitely need to do that.

> All in all, thanks for the initial implementation. I think we can merge this PR and consider if my comments are worth addressing in the future.
I will try to address some of them quickly as some of them seem very easy to do.